docs/getting_started.md

```diff
@@ -0,0 +1,246 @@
+
+!!! Note
+    Calling [`Application.register_action`][app_model.Application.register_action] with a single
+    [`Action`][app_model.Action] object is just a convenience around independently registering
+    objects with each of the registries using:
+
+    - [CommandsRegistry.register_command][app_model.registries.CommandsRegistry.register_command]
+    - [MenusRegistry.append_menu_items][app_model.registries.MenusRegistry.append_menu_items]
+    - [KeyBindingsRegistry.register_keybinding_rule][app_model.registries.KeyBindingsRegistry.register_keybinding_rule]
+
+### Registry events
+
+Each of these registries has a signal that is emitted when a new item is added.
+
+- `CommandsRegistry.registered` is emitted with the new command id (`str`) whenever
+  [`CommandsRegistry.register_command`][app_model.registries.CommandsRegistry.register_command]
+  is called
+- `MenusRegistry.menus_changed` is emitted with the new menu ids (`set[str]`) whenever
+  [`MenusRegistry.append_menu_items`][app_model.registries.MenusRegistry.append_menu_items]
+  or if the menu items have been disposed.
+- `KeyBindingsRegistry.registered` is emitted (no arguments) whenever
+  [`KeyBindingsRegistry.register_keybinding_rule`][app_model.registries.KeyBindingsRegistry.register_keybinding_rule] is called.
+
+You can connect callbacks to these events to handle them as needed.
+
+```python
+@my_app.commands.registered.connect
+def on_command_registered(command_id: str):
+    print(f'Command {command_id!r} registered!')
+
+my_app.commands.register_command('new-id', lambda: None, title='No-op')
+# Command 'new-id' registered!
+```
+
+## Executing Commands
+
+Registered commands may be executed on-demand using [`execute_command`][app_model.registries.CommandsRegistry.execute_command] method on the command registry:
+
+```python
+my_app.commands.execute_command('open')
+# prints "open file!" from the `open_file` function registered above.
+```
+
+### Command Arguments and Dependency Injection
+
+The `execute_command` function does accept `*args` and `**kwargs` that will
+be passed to the command.  However, very often in a GUI application
+you may wish to infer some of the arguments from the current state of the
+application.  For example, if you have menu item linked to a "close window",
+you likely want to close the current window. For this, `app-model` uses
+a dependency injection pattern, provided by the
+[`in-n-out`](https://github.com/pyapp-kit/in-n-out) library.
+
+The application has a [`injection_store`][app_model.Application.injection_store]
+attribute that is an instance of an `in_n_out.Store`.  A `Store` is a collection
+of "providers" (functions that can be called to return an instance of a given
+type), and "processors" (functions that accept an instance of a given type and
+do something with it). The `Store` is used to provide the arguments to the
+command when it is executed, based on the type annotations in the command
+function definition.
```

Comment:
nitpick

```
A `Store` is a collection
of "providers" (functions that can be called to return an instance of a given
type), and "processors" (functions that accept an instance of a given type and
do something with it). The `Store` is used to provide the arguments to the
command when it is executed, based on the type annotations in the command
function definition.
```

Nitpick:
```
A `Store` is a collection of: 

* "providers": functions that can be called to return an instance of a given type. This is used provide the arguments to the command when it is executed, based on the type annotations in the command
function definition.
* "processors": functions that accept an instance of a given type and do something with it. This is used process
   the return value of the command function at execution time, based on command definition return type annotations.
```

Reply:
Might feel redundant but a processor example below could be nice.

Reply:
good one. thanks added